src/util/scoped_timer.cpp

```diff
@@ -218,7 +228,10 @@ struct scoped_timer::imp {
             if (!init)
                 sched_yield();
         }
+        pthread_mutex_lock(&m_mutex);
```

Comment:
this mutex protects both m_signal_sent as well as the condition variable. THe only code that reads m_signal_sent seems within the same mutex.